backends-common/postgres/src/main/java/org/apache/james/backends/postgres/utils/PostgresExecutor.java

```diff
@@ -115,6 +115,12 @@ public Mono<Record> executeRow(Function<DSLContext, Publisher<Record>> queryFunc
                 .filter(preparedStatementConflictException()));
     }
 
+    public Mono<Optional<Record>> executeSingleRowOptional(Function<DSLContext, Publisher<Record>> queryFunction) {
```

Comment:
I'm not against it. 
But I don't know why we need it, We can completely handle it with SwitchIfEmpty function, at least in `retrieveVacation` method

Reply:
Cassandra has that util method too. I think we can keep it.